src/pages/change-password/index.tsx

```diff
@@ -1,65 +1,84 @@
   const user = useUser()
-  const [password, setPassword] = useState<string>("")
-  const [name, setName] = useState<string>("")
-  const [confirmPassword, setConfirmPassword] = useState<string>()
-  const [errorText, setErrorText] = useState<string>("")
-  const [showErrorText, setShowErrorText] = useState<boolean>(false)
   const router = useRouter()
+  const [errorMsg, setErrorMsg] = useState("")
+  const [showError, setShowError] = useState(false)
+  const {
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm()
+  const password = watch("password")
+  const name = watch("userName")
 
-  const setError = (errorText: string) => {
-    setErrorText(errorText)
-    setShowErrorText(true)
-  }
-
+  const changePasswordFormData: FormDataItem[] = [
+    {
+      tag: "userName",
+      type: "text",
+      placeholder: "Username",
+      dataCy: "set-name-textfield",
+      validationOptions: { required: "Username is required" },
+    },
+    {
+      tag: "password",
+      type: "password",
+      placeholder: "Password",
+      dataCy: "change-password-textfield",
+      validationOptions: {
+        required: "Password is required",
+        minLength: { value: 6, message: "Password should at least be 6 characters." },
+      },
+    },
+    {
+      tag: "confirmPassword",
+      type: "password",
+      placeholder: "Confirm Password",
+      dataCy: "change-password-confirm-textfield",
+      validationOptions: {
+        required: "Confirm Password is required",
+        minLength: { value: 6, message: "Password should be minimal 6 characters" },
+        validate: (value) => value === password || "Passwords do not match",
+      },
+    },
+  ]
   const submitNewPassword = async () => {
-    if (!password) {
-      setError("Please enter a password.")
-      return
-    }
+    if (!user) return
+    const { data, error } = await supabase.auth.updateUser({ password })
 
-    if (password.length < 6) {
-      setError("Password should at least be 6 characters.")
-      return
-    }
-
-    if (password !== confirmPassword) {
-      setError("Fill in equal passwords.")
-      return
-    }
-
-    if (name.length < 1) {
-      setError("Please fill in a name.")
-      return
-    }
-
-    if (user) {
-      const { data, error } = await supabase.auth.updateUser({ password })
-      if (error) {
-        // TODO Alert that change password failed, try again
-        setError(error.message)
-      } else if (data) {
-        setNameAndFirstLoginFalse(supabase, user.id, name)
-          .then((response) => {
-            const { error } = response
-            // TODO Error handling
-            if (error) {
-              setError(error.message)
-            } else router.push("/")
+    if (error) {
+      setErrorMsg("Change password failed, try again")
+      setShowError(true)
+    } else if (data) {
+      setNameAndFirstLoginFalse(supabase, user.id, name).then((response) => {
```

Comment:
My bad, but could you change the name of this method to setName since we don't have is_first_login anymore?
🙏🏾